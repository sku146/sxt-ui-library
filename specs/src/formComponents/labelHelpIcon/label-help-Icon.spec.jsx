import React from 'react';
import renderer from 'react-test-renderer';
import LabelHelpIcon from 'src/formComponents/labelHelpIcon/label-help-Icon.jsx';

describe("LabelHelpIcon", () => {

  let defaultProps = {
    name: 'test label name',
    text: 'label text',
    helpHeader: 'Help',
  };

  const RendererComponent = (props) => {
    return renderer.create(<LabelHelpIcon {...props}><div>Children test</div></LabelHelpIcon>);
  };

  it('component should render as expected', ()=> {
    const component = RendererComponent(defaultProps);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('component should render as expected with helpContent', ()=> {
    defaultProps.helpContent = 'In development';
    const component = RendererComponent(defaultProps);
    expect(component.toJSON()).toMatchSnapshot();
  });
});
