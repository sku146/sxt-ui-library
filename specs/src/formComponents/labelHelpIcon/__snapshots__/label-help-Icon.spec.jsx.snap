// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelHelpIcon component should render as expected 1`] = `
<div
  className="label-help-icon"
>
  <label
    className="css-label"
    data-bdd-selector="test label nameLabel"
    htmlFor="test label name"
  >
    label text
  </label>
  <div>
    Children test
  </div>
</div>
`;

exports[`LabelHelpIcon component should render as expected with helpContent 1`] = `
<div
  className="label-help-icon"
>
  <label
    className="css-label"
    data-bdd-selector="test label nameLabel"
    htmlFor="test label name"
  >
    label text
  </label>
  <i
    aria-hidden="true"
    className="help circle outline icon"
    data-bdd-selector="test label nameHelpIcon"
  />
  <div>
    Children test
  </div>
</div>
`;
